docs/src/reference/editions.md

```diff
@@ -0,0 +1,40 @@
+## Editions
+
+Editions of buffrs mark a specific evolutionary state of the package manager.
+The edition system exists to allow for fast development of buffrs while
+allowing you to already migrate existing protobufs to buffrs even though it
+has not yet reached a stable version.
+
+Editions can be either explicitly stated in the `Proto.toml` or are
+automatically inlined once a package is created using buffrs. This ensures that
+you dont need to care about them as a user but get the benefits.
+
+> Note: If you release a package with an edition that is incompatible with
+> another one (e.g. `0.7` is incompatible with `0.8`) you will need to
+> re-release the package for the new edition (by bumping the version, or
+> overriding the existing package) to regain compatibility.
+
+You may see errors like this if you try to consume (or work on) a package of
+another edition.
+
+```
+Error:   × could not deserialize Proto.toml
+  ╰─▶ TOML parse error at line 1, column 1
+        |
+      1 | edition = "0.7"
+        | ^^^^^^^^^^^^^^^
+      unsupported manifest edition, supported editions of 0.8.0 are: 0.8
+```
+
+### Canary Editions
+
+```toml
+edition = "0.7"
+```
+
+Canary editions are short-lived editions that are attached to a specific
+minor release of buffrs in the `0.x.x` version range. The edition name contains
+the minor version it is usable for. E.g. the edition `0.7` is usable /
+supported by all `0.7.x` buffrs releases. Compatibility beyond minor releases
+is not guaranteed as fundamental breaking changes may be introduced between
+editions.
```

Comment:
Why the special case for `0.x` versions and calling them Canary? [SemVer](https://semver.org/#spec-item-4) explicitly allows breaking changes pre `1.0` release, we don't have to special-case it.

I understand we want a mechanism for differentiating between minor and braking changes, but that's what major and minor versions are for. We are redefining semantic versioning (`major -> minor, minor -> patch`), causing potential confusion, for the questionable gain of keeping the major version below `1`?

So we either release `1.0` today, and then can release non-breaking changes as `1.y` and breaking changes as `x.0`, or we stick with `0.y` and live with the fact that not every minor release breaks things, while each requires updating buffrs packages.

Finally, a nitpick, but I think the word `Canary` is misused here. My understanding of [canary release](https://martinfowler.com/bliki/CanaryRelease.html) is to deploy a new version for only a small subset of consumers while monitoring for potential issues. We are not doing that, instead introducing a new version, while not forcing it on users, and with no intention of reverting. May I propose `Minor Editions` or `Patch Editions`, as that's literally what they are?

Reply:
So (ccing @xfbs) the meaning behind canary in the software comes from canary birds used to detect gas in coal mines, so translated this means "may be dangerous", hence canary editions of buffrs are intended to be used by early adopters that are ok with hitting issues.

Regarding the Edition system, please see the thread in slack with @Tpt - these editions are not meant to be supported longterm, but serve as a way to detect when / and with which buffrs version a package was released. This is independent from the semver meaning.

Reply:
+1 from me for getting canaries involved into even more software engineering :smile:

Reply:
```
Canary release is a technique to reduce the risk of introducing a new software version in production by slowly rolling out the change to a small subset of users before rolling it out to the entire infrastructure and making it available to everybody. 
```

I still don't think we're using the word in the right context here and I'm worried the meaning-overloading will be confusing to users familiar with the above definition. Still, this is a nit, so feel free to resolve this thread.

Reply:
Just as a hint: the canary system is internal and not necessarily user facing, also editions are automated and not necessarily managed by the user itself.